DPGAnalysis/Skims/python/ZMuSkim_cff.py

```diff
@@ -1,44 +1,166 @@
+looseMuonsForZMuSkim = cms.EDFilter("TrackSelector",
+                             src = cms.InputTag("generalTracks"),
+                             cut = cms.string('pt > 10 &&  abs(eta)<2.4 &&  (charge!=0)'),
                              filter = cms.bool(True)                                
                              )
 
-tightMuonsForZMuSkim = cms.EDFilter("MuonSelector",
-                             src = cms.InputTag("looseMuonsForZMuSkim"),
-                             cut = cms.string('pt > 20'),
-                             filter = cms.bool(True)                                
+
+
+###cloning the previous collection into a collection of candidates
+ConcretelooseMuonsForZMuSkim = cms.EDProducer("ConcreteChargedCandidateProducer",
+                                              src = cms.InputTag("looseMuonsForZMuSkim"),
+                                              particleType = cms.string("mu+")
+                                              )
+
+
+
+###create iso deposits
+tkIsoDepositTk = cms.EDProducer("CandIsoDepositProducer",
+                                src = cms.InputTag("ConcretelooseMuonsForZMuSkim"),
+                                MultipleDepositsFlag = cms.bool(False),
+                                trackType = cms.string('track'),
+                                ExtractorPSet = cms.PSet(
+        #MIsoTrackExtractorBlock
+        Diff_z = cms.double(0.2),
+        inputTrackCollection = cms.InputTag("generalTracks"),
+        BeamSpotLabel = cms.InputTag("offlineBeamSpot"),
+        ComponentName = cms.string('TrackExtractor'),
+        DR_Max = cms.double(0.5),
+        Diff_r = cms.double(0.1),
+        Chi2Prob_Min = cms.double(-1.0),
+        DR_Veto = cms.double(0.01),
+        NHits_Min = cms.uint32(0),
+        Chi2Ndof_Max = cms.double(1e+64),
+        Pt_Min = cms.double(-1.0),
+        DepositLabel = cms.untracked.string('tracker'),
+        BeamlineOption = cms.string('BeamSpotFromEvent')
+        )
+                                )
+
+###adding isodeposits to candidate collection
+allPatTracks = patGenericParticles.clone(
+    src = cms.InputTag("ConcretelooseMuonsForZMuSkim"),
+    # isolation configurables
+    userIsolation = cms.PSet(
+      tracker = cms.PSet(
+        veto = cms.double(0.015),
+        src = cms.InputTag("tkIsoDepositTk"),
+        deltaR = cms.double(0.3),
+        #threshold = cms.double(1.5)
+      ),
+      ),
+    isoDeposits = cms.PSet(
+        tracker = cms.InputTag("tkIsoDepositTk"),
+        ),
+    )
+
+
+
+
+###create the "probe collection" of isolated tracks 
+looseIsoMuonsForZMuSkim = cms.EDFilter("PATGenericParticleSelector",  
+                             src = cms.InputTag("allPatTracks"), 
+                             cut = cms.string("(userIsolation('pat::TrackIso')/pt)<0.4"),
+                             filter = cms.bool(True)
                              )
 
+
+
+
+
+MyPatMuons = cms.EDProducer("PATMuonProducer",
```

Comment:
please pick a more descriptive name for your producer (presumably one connected with the skim name)